honeybee_grasshopper_energy/src/HB Blind Material.py

```diff
@@ -0,0 +1,92 @@
+# Honeybee: A Plugin for Environmental Analysis (GPL)
+# This file is part of Honeybee.
+#
+# Copyright (c) 2019, Ladybug Tools.
+# You should have received a copy of the GNU General Public License
+# along with Honeybee; If not, see <http://www.gnu.org/licenses/>.
+# 
+# @license GPL-3.0+ <http://spdx.org/licenses/GPL-3.0+>
+
+"""
+Create a material for a blind layer in a window construction.
+This material can be plugged into the "HB Window Construction" component.
+_
+Window blind properties consist of flat, equally-spaced slats.
+-
+
+    Args:
+        _name: Text string for material name.
+        _vertical_: Set to "True" to have the blinds be vertically-oriented and
+            set to "False" to have them be horizontally-oriented.
+            Default: "False" for horizontal.
+        _slat_width_: The width of slat measured from edge to edge [m].
+            Default: 0.025 m (25 mm).
+        _slat_separation_: The distance between each of the slats [m].
+            Default: 0.01875 m (18.75 mm).
+        _slat_thickness_: A number between 0 and 0.1 for the thickness of the slat [m].
+            Default: 0.001 m (1 mm).
+        _slat_angle_: A number between 0 and 180 for the angle between the slats
+            and the glazing normal in degrees. 90 signifies slats that are
+            perpendicular to the glass. Default: 45.
```

Comment:
I think this component has too many inputs and maybe we can group some of them into another component. Maybe we should come up with a maximum number of inputs for best practice.

Reply:
If it makes you feel any better, the number of inputs here has already been halved from the number of inputs to the EnergyPlus object.  Also, the component is significantly smaller than its legacy cousin (the "Window Shade Generator") as you can see here:
![image](https://user-images.githubusercontent.com/5567574/70555723-9f768d80-1b4d-11ea-86d7-6979428b5312.png)

So I have already done a lot of work combining inputs that people might want to change together (like visible/solar reflectance or opening multipliers).  If people want to change those grouped inputs individually, they can just use the Python API to edit the material. I have also weeded out the inputs that I think are too advanced (like max and min slat angle).

All of this is to say that I would need a specific suggestion about what inputs should be removed or grouped with others if you want me to chop down this component further.

Reply:
To be more specific I think `slate` can be its own component which has orientation, width, separation, thickness and angle.

Reply:
I don't think that distinction is a good one because, when I eventually get around to implementing dynamic window constructions, I see all of the properties currently exposed as part of the "state". Essentially, the way that you make dynamic EnergyPlus constructions is by creating two separate constructions (an "on" and an "off" state) and then describing the control logic for switching between the two. Building dynamic constructions this way gives the user a lot more control over what they can simulate (like modeling a blind that changes both its angle and reflectance in response to a control input).  I always lamented that we didn't set up the legacy components to enable this wider range of modeling dynamic constructions and I intend to correct it in Honeybee[+].

So separating out some inputs that are the "state" and others that are not a part of the state doesn't seems like I'm going back to the world of honeybee-legacy and its unnecessary restrictions.  I'd be more open to picking some inputs between this component and the "Shade Material" component and pulling them out into another component or removed altogether.

Reply:
You know what fits better here. As long as it is not a single component with 13 inputs it should work.

Reply:
Are the 13 inputs really a problem, though?  I mean, the Ladybug[+] Import EPW currently has 16 outputs but the size of that component doesn't seem to be an issue.  Maybe if you could clarify why a component of this size is an issue, I can better decide what to leave out or how to address your concern. If the issue is that this component has a lot of inputs while the Import EPW has a lot of outputs and one might worry about a user having to specify a lot of inputs to make a blind material, I'd say that all of the inputs to this component except one have defaults and so users only need to change the properties that are relevant.  Or is the issue that this blind material is a rarely used component while the Import EPW is a common one? If anything, it seems like rarer components might be able to get away with bending rules more than the common ones.

Reply:
We agreed to leave this as is for now but we will play it by ear if people find the large number of inputs overwhelming.  And I came to a better understanding about Mostapha's concerns, which mean that any component with a larger number of inputs should definitely be using plenty of defaults in order to make the use of the component less overwhelming.